### PR TITLE
Compatibility with Avro v1.10.0

### PR DESCRIPTION
**Pull request by tjwp**
Extend compatibility to Avro v1.10.0 -- no changes required.

Update the Ruby version for the repo and the versions that the gem is tested against.

### CODE REVIEW

File: avro-patches.gemspec [hunk 1]
```diff
@@ -31,5 +31,5 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'test-unit'
   spec.add_development_dependency 'overcommit'
 
-  spec.add_runtime_dependency 'avro', '>= 1.9.0', '<= 1.9.1'
+  spec.add_runtime_dependency 'avro', '>= 1.9.0', '<= 1.10.0'
```
Comment (jturkel) [inline avro-patches.gemspec]: Should we be more optimistic now that this gem has no code and remove the upper bound?

Comment (tjwp) [reply avro-patches.gemspec]: It needs to be bumped so infrequently that I'm okay with it being pessimistic. And I'm optimistic that at some point  more patches will appear here first!

But I don't feel that strongly either way.

Comment (jturkel) [reply avro-patches.gemspec]: I don't feel super strongly either but hopefully with the Avro project moving again we won't have any new patches ;) Think we should at least set an upper bound of `< 1.11`?

Comment (tjwp) [reply avro-patches.gemspec]: Yeah, I'm comfortable with 1.10.x.

**Review by jturkel — APPROVED** [review]

**Pull request merged**